push-sdk-swift/DeviceRegistration.swift

```diff
@@ -232,7 +245,7 @@ open class DeviceRegistration: NSObject, URLSessionTaskDelegate {
         } else {
             value = Bundle.main.object(forInfoDictionaryKey: key) as? String
         }
-        if (value == nil && value!.isEmpty)  {
+        if (value == nil || value!.isEmpty)  {
```

Comment:
@corinnekrych only this change is needed to fix the real problem, the other changes are to remove the assert and make register fail when some parameter is missing or empty instead of crashing the app.
WDYT?